File: src/com/anathema_roguelike/characters/inventory/PrimaryWeapon.java
package com.anathema_roguelike.characters.inventory;

import com.anathema_roguelike.characters.Character;
import com.anathema_roguelike.items.weapons.Weapon;
import com.anathema_roguelike.items.weapons.natural_weapons.Unarmed;

public class PrimaryWeapon extends SingleSlot<Weapon> {

	public PrimaryWeapon(Character character) {
		super(Weapon.class, character);
	}
	
	@Override
	protected Weapon getDefaultItem() {
		return new Unarmed();
	}
}

File: src/com/anathema_roguelike/entities/characters/inventory/PrimaryWeapon.scala


package com.anathema_roguelike
package entities.characters.inventory

import com.anathema_roguelike.entities.items.weapons.Weapon
import com.anathema_roguelike.entities.items.weapons.natural_weapons.Unarmed
import com.anathema_roguelike.entities.characters.Character

class PrimaryWeapon(character: Character) extends SingleSlot[Weapon](character) {
  override protected def getDefaultItem = new Unarmed(Right(character))
}